plugins/thumbnail/lib/model/imagickAction/kCropAction.php

```diff
@@ -25,8 +25,17 @@ protected function initParameterAlias()
 
 	protected function extractActionParameters()
 	{
-		$this->newWidth = $this->getIntActionParameter(kThumbnailParameterName::WIDTH);
-		$this->newHeight = $this->getIntActionParameter(kThumbnailParameterName::HEIGHT);
+		if($this->getIntActionParameter(kThumbnailParameterName::ROTATION) === 90 || $this->getIntActionParameter(kThumbnailParameterName::ROTATION) === 270)
```

Comment:
Shorter test - if($this->getIntActionParameter(kThumbnailParameterName::ROTATION)%180)